### PR TITLE
Add BBBPump.fun to xdc network

### PR DESCRIPTION
**Pull request by GalaxySciTech**
Name (to be shown on DefiLlama):

BBBPump

Twitter Link:

https://twitter.com/bbbpumpdotfun

List of audit links if any:

N/A

Website Link:

https://bbbpump.fun

Logo (High resolution, will be shown with rounded borders):

Please provide the high-resolution logo link, e.g., https://bbbpump.fun/favicon.ico

Current TVL:

Real-time TVL can be tracked here: https://xdcscan.com/address/0x2e24bfde1eeda0f1ea3e57ba7ff10ac6516ab5ec

Treasury Addresses (if the protocol has treasury):

N/A

Chain:

XDC Network

Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed):

(Leave blank if not listed)

Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed):

(Leave blank if not listed)

Short Description (to be shown on DefiLlama):

BBBPump is a DeFi protocol on the XDC Network offering liquidity mining and TVL management solutions.

Token address and ticker if any:

Category (full list at https://defillama.com/categories) *Please choose only one:

Yield

Oracle Provider(s):

N/A

Implementation Details:

The project calculates TVL using native XDC blockchain methods without integrating external oracles.

Documentation/Proof:

You can provide a link to the documentation or resources about BBBPump, e.g., 

forkedFrom (Does your project originate from another project):

N/A

methodology (what is being counted as tvl, how is tvl being calculated):

BBBPump’s TVL is calculated based on on-chain data from the XDC smart contract address 0x2e24bfde1eeda0f1ea3e57ba7ff10ac6516ab5ec, including all deposited liquidity assets.

Github org/user (Optional, if your code is open source, we can track activity):

### CODE REVIEW

**Comment by llamabutler** [comment]
The adapter at projects/bbbpump exports TVL: 
        
 
 
    xdc                       0
    
    total                    0 
    
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'

**Comment by GalaxySciTech** [comment]
> The adapter at projects/bbbpump exports TVL:
> 
> ```
> xdc                       0
> 
> total                    0 
> 
> Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
> ```

it seems some error , could u help me write correctly code pls

**Comment by llamabutler** [comment]
The adapter at projects/bbbpump exports TVL: 
        
 
 
    xdc                       73.62 k
    
    total                    73.62 k

**Comment by llamabutler** [comment]
The adapter at projects/bbb exports TVL: 
        
 
 
    xdc                       0
    
    total                    0

**Comment by llamabutler** [comment]
The adapter at projects/bbbpump exports TVL: 
        
 
 
    xdc                       73.40 k
    
    total                    73.40 k

**Comment by llamabutler** [comment]
The adapter at projects/bbb exports TVL: 
        
 
 
    xdc                       0
    
    total                    0

**Comment by llamabutler** [comment]
The adapter at projects/bbbpump exports TVL: 
        
 
 
    xdc                       73.14 k
    
    total                    73.14 k

File: projects/bbbpump/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const { sumTokensExport,nullAddress } = require('../helper/unwrapLPs');
+module.exports = {
+    start: '2024-10-10',
+    xdc: { tvl: sumTokensExport({ owners: [
+        '0x2E24BFdE1EEDa0F1EA3E57Ba7Ff10ac6516ab5Ec', // BBBPump
+        '0x37c00AE5C4b49Ab0F5fD2FFB1033588e9bC33B08', // Megadrop BBB
+      ], tokens: [
+        nullAddress,
+        "0xfa4ddcfa8e3d0475f544d0de469277cf6e0a6fd1"
```
Comment (g1nt0ki) [inline projects/bbbpump/index.js:9]: what is this token? BBB? why are you counting it as tvl? 

staked own tokens can be counted under staking: https://docs.llama.fi/list-your-project/what-to-include-as-tvl

File: projects/bbb/index.js
Comment (g1nt0ki) [inline projects/bbb/index.js:1]: dont understand why you are trying to counting token supply as tvl?

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
.

File: projects/bbbpump/index.js (hunk 1 above)
Comment (GalaxySciTech) [reply projects/bbbpump/index.js:9]: fixed

**Comment by llamabutler** [comment]
The adapter at projects/bbbpump exports TVL: 
        
 
 
    xdc                       71.06 k
    
    total                    71.06 k

**Comment by GalaxySciTech** [comment]
tvl need show bbb

**Comment by llamabutler** [comment]
The adapter at projects/bbbpump exports TVL: 
        
 
 
    xdc                       70.95 k
    xdc-staking               0
    staking                   0
    
    total                    70.95 k

**Comment by llamabutler** [comment]
The adapter at projects/bbbpump exports TVL: 
        
 
 
    xdc-staking               4.88 M
    staking                   4.88 M
    xdc                       70.19 k
    
    total                    70.19 k

**Comment by llamabutler** [comment]
The adapter at projects/bbbpump exports TVL: 
        
 
 
    xdc-staking               4.79 M
    staking                   4.79 M
    xdc                       70.26 k
    
    total                    70.26 k

**Comment by llamabutler** [comment]
The adapter at projects/bbbpump exports TVL: 
        
 
 
    xdc-staking               4.76 M
    staking                   4.76 M
    xdc                       69.42 k
    
    total                    69.42 k

**Comment by g1nt0ki** [comment]
looks good thanks

**Pull request merged**